### PR TITLE
mgr/dashboard: carbon datatable minor bug fixes

### PR DESCRIPTION
**Pull request by nizamial09**
- fixes empty white spaces in the table
- fixes rgw action missing the dropdown which has more action
- fixes some tables detail component where it didn't had carbon table
- fix the multi-site details view table not showing up
- fix the nvmeof modals like namespace deletion not getting opened
- fix the cached reload button status not showing correctly in rgw table
- fix show button in the rgw users detail page broken
- fix the multi-cluster list component issue where details component is
  not following carbon attributes
- fixing the configuration page where it does a page reload on form
  submit. instead doing a table refresh

- additionally fixed a validation issue in the service form

**BEFORE**
empty space issue
Screenshot from 2024-08-29 11-41-15

 rgw action missing the dropdown which has more action
Screenshot from 2024-08-29 11-42-34

some table details without carbon
Screenshot from 2024-08-29 11-43-10

Multisite details view not showing up
Screenshot from 2024-08-29 11-41-33

**AFTER**
empty space
image

rgw missing actions fixed
Screenshot from 2024-08-29 11-58-14

fixed carbon table details
Screenshot from 2024-08-29 11-43-03

multisite details view
image

Fixes: https://tracker.ceph.com/issues/67788
Fixes: https://tracker.ceph.com/issues/67552

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [x] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by afreen23 — APPROVED** [review]
Thanks @nizamial09 works fine , tested this.

**Review by ivoalmeida — APPROVED** [review]
lgtm

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-form/service-form.component.ts [hunk 1]
```diff
@@ -195,7 +195,12 @@ export class ServiceFormComponent extends CdForm implements OnInit {
           })
         ]
       ],
-      group: [null, Validators.required],
+      group: [
+        null,
+        CdValidators.requiredIf({
+          service_type: 'nvmeof'
+        })
+      ],
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-form/service-form.component.ts:203]: Thanks for fixing this.
This is breaking service form now for other components FYI
lets add this in description too

Comment (nizamial09) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-form/service-form.component.ts:203]: yup, its there btw.

> additionally fixed a validation issue in the service form

**Comment by afreen23** [comment]
Some lint errors:
```[lint:html    ] 
[lint:html    ] [htmllint] found 0 errors out of 243 files
[lint:html    ] All files pass linting
[lint:prettier] cypress/e2e/multi-cluster/multi-cluster.po.ts
[lint:prettier] cypress/e2e/page-helper.po.ts
[lint:eslint  ] 
[lint:eslint  ] Linting "ceph-dashboard"...
[lint:eslint  ] 
[lint:eslint  ] All files pass linting.
[lint:eslint  ] 

Try running 'npm run fix' to fix some linting errors. Some errors might need a manual fix.
```

**Review by cloudbehl — APPROVED** [review]

**Comment by nizamial09** [comment]
the multi-cluster e2e failure is unrelated to this PR and I am trying to fix it separately: https://github.com/ceph/ceph/pull/59538

**Comment by nizamial09** [comment]
merging this since this fixes a lot of UI issues along with some test failures.

**Pull request merged**